src/js/jsx/sections/libraries/assets/AssetSection.jsx

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (c) 2014 Adobe Systems Incorporated. All rights reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ *
+ */
+
+define(function (require, exports, module) {
+    "use strict";
+
+    var React = require("react"),
+        Fluxxor = require("fluxxor"),
+        FluxMixin = Fluxxor.FluxMixin(React);
+        
+    var strings = require("i18n!nls/strings");
+    
+    var SplitButton = require("jsx!js/jsx/shared/SplitButton"),
+        SplitButtonList = SplitButton.SplitButtonList,
+        SplitButtonItem = SplitButton.SplitButtonItem;
+
+    var AssetSection = React.createClass({
+        mixins: [FluxMixin],
+
+        propTypes: {
+            element: React.PropTypes.object.isRequired,
+            onSelect: React.PropTypes.func
+        },
+
+        componentDidMount: function () {
+            this._listenTitleEvent();
+        },
+        
+        componentDidUpdate: function () {
+            this._listenTitleEvent();
+        },
+        
+        /**
+         * Listen to the title's double click event for renaming purpose.
+         *
+         * @private
+         */
+        _listenTitleEvent: function () {
+            if (!this.refs.assetTitle) {
+                return;
+            }
+            
+            var assetTitleNode = React.findDOMNode(this.refs.assetTitle);
+            assetTitleNode.addEventListener("dblclick", this._handleRename);
+        },
+
+        /**
+         * Handle select asset event.
+         * @param  {SynthenticEvent} event
+         *
+         * @private
+         */
+        _handleSelect: function (event) {
+            if (event.target !== React.findDOMNode(this)) {
```

Comment:
This seems weird as well.. clickEvent should only come here if this component is indeed the target. If you're getting click events from one of it's children, you probably want to stopPropagation in the children's click handlers.